samples: mesh: nrf52: removed unwanted code

vikrant8052 opened the pull request:
Removed unwanted code which is not allowing GET response
to get send by Server to client. After removing this, there
is improvement in GATT Bearer performance while testing
with Android nRFMesh App.

Signed-off-by: Vikrant More <vikrant8051@gmail.com>

vikrant8052 committed:
samples: mesh: nrf52: removed unwanted code

Removed unwanted code which is not allowing GET response
to get send by Server to client. After removing this, there
is improvement in GATT Bearer performance while testing
with Android nRFMesh App.

Signed-off-by: Vikrant More <vikrant8051@gmail.com>

vikrant8051 commented:
Please wait.

jhedberg commented on samples/boards/nrf52/mesh/onoff_level_lighting_vnd_app/src/mesh/device_composition.c:
```diff
@@ -101,6 +101,31 @@ static struct bt_mesh_elem elements[];
 
 /* message handlers (Start) */
 
+#define common()							       \
```
There are many things wrong with this:

 - It should be a helper function
 - It should have a descriptive name matching what it does
 - Even if it was a macro it should be all upper-case

vikrant8051 replied:
Removed that macro. So issue has solved !!

vikrant8052 committed:
samples: mesh: nrf52: reassignment of last target values

Reassignment of last target values as per present values
after NODE reset as well as after every transition
get completed.

Signed-off-by: Vikrant More <vikrant8051@gmail.com>

vikrant8052 committed:
samples: mesh: nrf52: removed bug introduced by PR:9521

Removed bug which get introduced by PR:9521. Because of it
not able to save App's variable on SoC flash.

Signed-off-by: Vikrant More <vikrant8051@gmail.com>

vikrant8051 commented:
@jhedberg 
After making changes as per 3rd commit, I could save #ResetCounter value on flash. Otherwise on every reboot it is not getting updated & remains Zero. That means changes made by https://github.com/zephyrproject-rtos/zephyr/pull/9521 have issue.
